### ISSUE TITLE
tf.keras.layers.Softmax should respect image_data_format by default

### ISSUE THREAD

**Issue by dd1923** [tensorflow/tensorflow#36679]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.1.0
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
Tensorflow allows one to set image_data_format of either channels_last (set by default, NHWC) or channels_first (NCHW) using `tf.keras.backend.set_image_data_format(data_format)`. Doing so nicely results in almost all layers of the `tf.keras.model` adjust appropriately by either bringing the channel last or first. Even the model.load_weights() work perfectly fine (and transparently), respecting user's choice of image_data_format, as it should. The story is somewhat different when it comes to `tf.keras.layers.Softmax`. Softmax layer/activation is often used in classification problems to normalize the output of the model and such normalization is applied w.r.t to an axis. Currently, Softmax implementation always sets the default value of axis = -1, which is the last axis. This works perfectly fine when image_data_format is channels_last but it for channels_first, one needs to explicitly set the axis to 1. 

This feels like slightly out of sync compared to how all the other layers behave. They respect the image_data_format automatically and adjust accordingly. So why not Softmax? I am pretty sure 90% of the use cases of Softmax applies the function along the channels axis, so why not make it respect the data format automatically,too ?

This is also easy to miss for folks just getting started with TensorFlow. The docs does not mention anything remotely about this.

**Will this change the current api? How?**
Yes. `tf.keras.layers.Softmax` would start respecting the default set_image_data_format set by user and hence axis would be -1 incase of NHWC and 1 incase of NCHW.

This would also allow one to easily specify `activation="softmax"` in the last layer instead of adding one explicit line like :

`model.add(tf.keras.layers.Softmax(axis=-1 if data_format=="channels_last" else 1))`

**Who will benefit with this feature?**
A big group of people who are either using Softmax in its default form (i.e. to normalize along channels axis) or someone who is getting started with TF and wants to have flexibility of changing image data layouts. This would render a uniform API experience and expectation to end-user, trusting that tensorflow's layers are respecting the global image data format without much hassle or individual treatment.

**Any Other info.**

**Comment by dd1923** [comment]
Does softmax now work with channels last and first?

**Comment by fchollet** [comment]
Just seeing this now -- the reason why `Softmax` cannot default to a different `axis` value based on `image_data_format` is that it isn't an image-processing layer (unlike `Conv2D`), and it has no way of knowing whether the tensor it's processing represents image features or not.

A rule like "if the input tensor is 4D, and `image_data_format == 'channels_first'`, then use `axis=1`" would be very error-prone. It is much better to let the user specify their own `axis`value.

**Comment by dd1923** [comment]
Can it raise a warning instead? I am sure for most users would want to have axis = 1 when image_data_format is channels first. If they don't specify an axis and if image_data_format is channels first, can it raise a warning or message so that the user is atleast aware?
